### DISCUSSION TITLE
iOS Silent notifications not showing when the app is killed

### DISCUSSION THREAD

**Discussion by MatthieuPeyrot** [invertase/react-native-firebase#4747] [Q&A] [answered]
Hi,

We implemented the notification with react native firebase and notifee, work well on android ( foreground , background, killed) but for iOS it works well for foreground and background but we didn't manage to make it work on the killed part.

We use silent notification to be able to change it with notifee to add attachement and action buttons.

I can see on my device log that i have an incoming notification but it's like it doesn't go trigger the app and the `setBackgroundMessageHandler` function.

I got also in the logs this line `no data found .... at PendingNotifications.plist`

We tried with `sendToDevice` and `send` on the backend and with `content-available : 1` , hight priority 

Does someone have encounter this issue and find any solution ?

**Comment by mikehardy** [comment, marked as answer]
This is very common. If you look at issues and discussions here (please do) you will notice tens maybe hundreds have encountered the issue, and it's been discussed a lot.

TL;DR: iOS background data-only notifications are unreliable for a variety of reasons. -FIRDebugEnabled and watching log on real device will show them discarding sometimes, do that. If it's delivered but the app doesn't fire, iOS just chose not to fire up the app. Nothing you can do.

Even more specifically if the user has swiped the app away (decisively killing it), iOS believes the user. The user said "don't run this app", so iOS won't.

**Comment by bourgois** [comment]
Well our issue isn't 'random' it is systematically not working... Plus it does work with other libraries.
Try testing with WhatsApp by example: it always get delivered when the app is killed. I doubt Apple made a special treatment for FB :)

**Reply by mikehardy** [reply]
I believe this is a misunderstanding of how the power miser algorithm works.
It learns you like whatsapp by carefully tracking your usage patterns of whatsapp, so whatsapp gets generous slices of power.
What has it learned about your app? Assuming that it is perfectly configured and the JSON you push is perfect, it may still have learned via interaction tracking that the pushes never receive interaction, so you get no power budget
